api/src/main/java/io/grpc/DoubleCounterMetricInstrument.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.util.List;
+
+/**
+ * Represents a double-valued counter metric instrument.
+ */
+@Internal
+public class DoubleCounterMetricInstrument extends PartialMetricInstrument  implements
+    MetricInstrument {
+  private final boolean isEnabledByDefault;
+
+  DoubleCounterMetricInstrument(long index, String name, String description, String unit,
+      List<String> requiredLabelKeys, List<String> optionalLabelKeys,
+      boolean isEnabledByDefault) {
+    super(index, name, description, unit, requiredLabelKeys, optionalLabelKeys);
+    this.isEnabledByDefault = isEnabledByDefault;
+  }
+
+  @Override
+  public long getIndex() {
+    return index;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
+
+  @Override
+  public String getUnit() {
+    return unit;
+  }
+
+  @Override
+  public List<String> getRequiredLabelKeys() {
+    return requiredLabelKeys;
+  }
+
+  @Override
+  public List<String> getOptionalLabelKeys() {
+    return optionalLabelKeys;
+  }
+
+  public boolean getEnabledByDefault() {
```

Comment:
Why isn't this part of MetricInstrument?

Reply:
Looking at https://github.com/grpc/grpc/pull/36324, I think the number of flags we want to add for each metric to denote stability is not final yet, so added it on instruments. Do you suggest moving it to `MetricInstrument`?

Reply:
Yeah, it is in flux, but it is in flux equally for all instrument types. When we update it, we will update all the instruments; it is independent.